cli/wallet/wallet.go

```diff
@@ -966,31 +977,54 @@ func ReadWalletConfig(configPath string) (*config.Wallet, error) {
 	return cfg, nil
 }
 
-func newAccountFromWIF(w io.Writer, wif string, scrypt keys.ScryptParams) (*wallet.Account, error) {
+func newAccountFromWIF(w io.Writer, wif string, scrypt keys.ScryptParams, label *string, pass *string) (*wallet.Account, error) {
+	var (
+		phrase, name string
+		err          error
+	)
+	if pass != nil {
+		phrase = *pass
+	}
+	if label != nil {
+		name = *label
+	}
 	// note: NEP2 strings always have length of 58 even though
 	// base58 strings can have different lengths even if slice lengths are equal
 	if len(wif) == 58 {
-		pass, err := input.ReadPassword(EnterPasswordPrompt)
-		if err != nil {
-			return nil, fmt.Errorf("Error reading password: %w", err)
+		if pass == nil {
+			phrase, err = input.ReadPassword(EnterPasswordPrompt)
+			if err != nil {
+				return nil, fmt.Errorf("error reading password: %w", err)
+			}
 		}
 
-		return wallet.NewAccountFromEncryptedWIF(wif, pass, scrypt)
+		acc, err := wallet.NewAccountFromEncryptedWIF(wif, phrase, scrypt)
+		if err != nil {
+			return nil, err
+		}
+		acc.Label = name
```

Comment:
So you won't ask for a NEP-2 name?

Reply:
Yes, it's done for compatibility with the old behaviour. Do we need to ask it if it's not provided via config?

Reply:
I think it's inconsistent, NEP-2 only differs from WIF in that it's encrypted. But new account should have a label. If we're in interactive mode it's OK to ask for a name, if we have it from a flag, it's not an issue already. And NEP-2 is not used very often for imports, to be fair, mostly people transfer keys using WIFs.

You can make it a separate patch though to be scheduled for 0.102.0.

Reply:
https://github.com/nspcc-dev/neo-go/pull/2889